module/VuFind/src/VuFind/Config/PathResolverFactory.php

```diff
@@ -82,14 +86,46 @@ public function __invoke(
         if (!empty($options)) {
             throw new \Exception('Unexpected options sent to factory.');
         }
-        $localDirs = defined('LOCAL_OVERRIDE_DIR')
+        $localDirs = [];
+        $currentDir = defined('LOCAL_OVERRIDE_DIR')
             && strlen(trim(LOCAL_OVERRIDE_DIR)) > 0
-                ? [
-                    [
-                        'directory' => LOCAL_OVERRIDE_DIR,
-                        'defaultConfigSubdir' => $this->defaultLocalConfigSubdir,
-                    ],
-                ] : [];
+            ? LOCAL_OVERRIDE_DIR : '';
+        while (!empty($currentDir)) {
+            // check if the directory exists
+            if (!($canonicalizedCurrentDir = realpath($currentDir))) {
+                trigger_error("Configured local directory does not exist: " . $currentDir, E_USER_WARNING);
+                break;
+            }
+            $currentDir = $canonicalizedCurrentDir;
+
+            // loading DirLocations.ini of currentDir
+            $systemConfigFile = $currentDir . '/DirLocations.ini';
+            $systemConfig = new Config(
+                file_exists($systemConfigFile)
+                    ? $this->getIniReader()->fromFile($systemConfigFile)
+                    : []
+            );
+
+            // adding directory to the stack
+            array_unshift(
+                $localDirs,
+                [
+                    'directory' => $currentDir,
+                    'defaultConfigSubdir' =>
+                        $systemConfig['Local_Dir']['config_subdir']
```

Comment:
The config file has a [LocalConfig] section, but the code is looking for [Local_Dir]. These should be reconciled.

Reply:
I must have missed that when I changed the section's name. Thanks for pointing it out! Fixed it.